Sources/AuthFoundation/Network/APIClient.swift

```diff
@@ -116,14 +157,59 @@ extension APIClient {
-                self.didSend(request: request, received: response)
+                guard let httpResponse = response as? HTTPURLResponse else {
+                    throw APIClientError.invalidResponse
+                }
                 
-                completion(.success(response))
+                let rateInfo = ApiRateLimit(with: httpResponse.allHeaderFields)
+                switch httpResponse.statusCode {
+                case 200..<300:
+                    let response: APIResponse<T> = try self.validate(data: data,
+                                                                     response: httpResponse,
+                                                                     rateInfo: rateInfo,
+                                                                     parsing: context)
+                    self.didSend(request: request, received: response)
+                    completion(.success(response))
+                case 429:
+                    guard let rateInfo = rateInfo else {
+                        fallthrough
+                    }
+                    let retry = state?.type ?? self.shouldRetry(request: request, rateLimit: rateInfo)
+                    switch retry {
+                    case .doNotRetry: break
+                    case .retry(let maximumRetryCount):
+                        let retryState: APIRetry.State
+                        if let state = state {
+                            retryState = state.nextState()
+                        } else {
+                            var requestId: String? = nil
+                            if let requestIdHeader = requestIdHeader {
+                                requestId = httpResponse.allHeaderFields[requestIdHeader] as? String
+                            }
+                            retryState = APIRetry.State(type: retry,
+                                                        requestId: requestId,
+                                                        originalRequest: request,
+                                                        retryCount: 1)
+                        }
+                        guard retryState.retryCount <= maximumRetryCount else {
+                            break
+                        }
+                        let urlRequest = addRetryHeadersToRequest(state: retryState)
+                        queue.asyncAfter(deadline: .now() + rateInfo.delay) {
+                            self.send(urlRequest, parsing: context, state: retryState, completion: completion)
+                        }
+                        return
+                    }
+                    fallthrough
+                default:
+                    if let error = error(from: data) ?? context?.error(from: data) {
+                        throw APIClientError.serverError(error)
+                    } else {
+                        throw APIClientError.statusCode(httpResponse.statusCode)
+                    }
+                }
```

Comment:
Shouldn't this be handled by existing code already? I'm not sure why more error handling for non-429 cases needs to be added here.

Reply:
This is the default fallthrough case.
```
-  if success code ... we handle accordingly 
- If 429 error we handle accordingly. 
- else : check the default case. 
```